### ISSUE TITLE
doc_markdown a bit too eager

### ISSUE THREAD

**Issue by Keats** [Manishearth/rust-clippy#902]
Running the latest clippy on https://github.com/Keats/tera/ gives the following warning:

```
Compiling tera v0.1.0 (file:///home/vincent/Code/tera)
src/lexer.rs:133:1: 133:94 warning: you should put `HxaD_trXwRE` between ticks in the documentation, #[warn(doc_markdown)] on by default
src/lexer.rs:133 /// Lexer based on the one used in go templates (https://www.youtube.com/watch?v=HxaD_trXwRE)
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lexer.rs:133:1: 133:94 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#doc_markdown

```

**Comment by mcarton** [comment]
IMHO it’s a feature here. Raw links are ugly.
The lint ignores actual markdown links `[go templates](https://www.youtube.com/watch?v=HxaD_trXwRE)` is OK.

**Comment by tikue** [comment]
It does not ignore markdown links:

```
$ cargo clippy
   Compiling tarpc v0.6.0 (file:///Users/tikue/Documents/rust/tarpc)
warning: you should put `gRPC` between ticks in the documentation, #[warn(doc_markdown)] on by default
  --> src/lib.rs:18:45
   |
18 | //! architectures. Two well-known ones are [gRPC](http://www.grpc.io) and
   |                                             ^^^^
```

**Comment by mcarton** [comment]
@tikue It's a feature. It ignores the link (`http://…`), not but the text (`gRPC`). Clippy cannot know every possible weird-cased name, so you can add

```toml
doc-valid-idents = ["gRPC"]
```

to `clippy.toml` to ignore it.

**Comment by tikue** [comment]
Gotcha. I think that's reasonable. Though I'd be interested to get some numbers on precision with this one...

**Comment by Keats** [comment]
Running into it again on a markdown link:

```bash
warning: you should put `Prevention_Cheat_Sheet` between ticks in the documentation, #[warn(doc_markdown)] on by default
 --> /home/vincent/Code/tera/src/utils.rs:3:76
  |
3 | /// From [OWASP](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet)
  |                                                                            ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#doc_markdown

```

**Comment by briansmith** [comment]
In https://github.com/briansmith/ring/issues/285#issuecomment-277328045, @pietro wrote:
>  BTW, currently clippy warns about lack of backticks on documentation [...]. Some of the things clippy warns about are definitively wrong, like SpecialPublications in http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf

Example:

```rust
/// See [NIST SP 800-56A, revision 2].
///
/// [NIST SP 800-56A, revision 2]:
///     http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf
fn foo() { }
```

**Comment by clarcharr** [comment]
@Keats that isn't a valid markdown link; you need to escape the \( \).

**Comment by sanmai-NL** [comment]
Could we please determine the default value of `doc-valid-idents` based on empirical data? 
A practical approach would be to include the top [StackOverflow tags](https://archive.org/download/stackexchange/stackoverflow.com-Tags.7z) that are proper names. Then we must normalize those to their canonical mixed case form, next add them (manually) to [`clippy_lints/src/utils/conf.rs`](clippy_lints/src/utils/conf.rs). This should only have to be done very infrequently.

**Comment by mcarton** [comment]
While I think it's a good idea how do you expect us to get such a list from a 47396-item list of lower-case IDs?

**Comment by sanmai-NL** [comment]
@mcarton:
I propose we match the top (the no. of entries doesn’t matter) StackOverflow tags against all mixed-case [Wikipedia English article titles](enwiki-latest-all-titles-in-ns0.gz). Mixed-case meaning, having uppercase letters after the initial. We can match pairs based on the lowest edit distance observed given a tag for each title
